src/PhpSpec/Process/Shutdown/UpdateConsoleAction.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpSpec\Process\Shutdown;
+
+use PhpSpec\Formatter\CurrentExampleWriter;
+use PhpSpec\Message\CurrentExample;
+
+final class UpdateConsoleAction implements ShutdownActionInterface
+{
+    /**
+     * @var CurrentExample
+     */
+    private $message;
+    /**
+     * @var CurrentExampleWriter
+     */
+    private $currentExampleWriter;
+
+    public function __construct(CurrentExample $message, CurrentExampleWriter $currentExampleWriter)
+    {
+        $this->message = $message;
+        $this->currentExampleWriter = $currentExampleWriter;
+    }
+
+    public function runAction()
+    {
+        ini_set('display_errors', '0');
+        error_reporting(E_ERROR);
+        register_shutdown_function(array($this, 'output'));
```

Comment:
This shouldn't be registering itself as a shutdown function, if we're going to have lots of `ShutdownActionInterface` instances.

Have one shutdown handler, that then calls all of the registered `ShutdownActionInterface`s